packages/mui-system/src/createStyled.js

```diff
@@ -219,7 +233,7 @@ export default function createStyled(input = {}) {
       if (process.env.NODE_ENV !== 'production') {
         let displayName;
         if (componentName) {
-          displayName = `${componentName}${componentSlot || ''}`;
+          displayName = `${componentName}${capitalize(componentSlot || '')}`;
```

Comment:
@mnajdova based on your [comment](https://github.com/mui/material-ui/pull/37908#issuecomment-1637938923), I think the logic can be the other way around.

I believe that using slotName directly is easier to understand for developers. The `capitalize` I added is to preserve the previous behavior. In fact, it can be removed because it is for debugging and it won't be used in production.

We could remove it in the next major release?

Reply:
I think we can keep using `capitalize` it in the `displayName`